src/Utilities/set_dot.jl

```diff
@@ -0,0 +1,352 @@
+# Copyright (c) 2017: Miles Lubin and contributors
+# Copyright (c) 2017: Google Inc.
+#
+# Use of this source code is governed by an MIT-style license that can be found
+# in the LICENSE.md file or at https://opensource.org/licenses/MIT.
+
+# Scalar product. Any vector set defined that does not use the standard scalar
+# product between vectors of ``R^n`` should redefine `set_dot` and
+# `dot_coefficients`.
+
+"""
+    set_dot(x::AbstractVector, y::AbstractVector, set::AbstractVectorSet)
+
+Return the scalar product between a vector `x` of the set `set` and a vector
+`y` of the dual of the set `s`.
+"""
+function set_dot(x::AbstractVector, y::AbstractVector, ::MOI.AbstractVectorSet)
+    return dot(x, y)
```

Comment:
```suggestion
    return LinearAlgebra.dot(x, y)
```
Let's make this explicit. I thought I'd removed `using LinearAlgebra`, but I guess I hadn't.